src/stream_builder.cpp

```diff
@@ -29,11 +29,7 @@ StreamBuilder::~StreamBuilder()
 {
     StreamBuilder *b = this;
     std::vector<StreamBuilder *> &builders = get_builders();
-    std::vector<StreamBuilder *>::iterator it = std::find(builders.begin(), builders.end(), b);
-    if (it != builders.end()) {
-        std::swap(*it, builders.back());
-        builders.pop_back();
-    }
+    builders.erase(std::remove(builders.begin(), builders.end(), b), builders.end());
```

Comment:
do you know if it will do any memmove between operations since you are erasing from start to end rather than from end to beginning?